codec/src/main/java/io/netty/handler/codec/ByteToMessageDecoder.java

```diff
@@ -263,14 +299,20 @@ public final void handlerRemoved(ChannelHandlerContext ctx) throws Exception {
      */
     protected void handlerRemoved0(ChannelHandlerContext ctx) throws Exception { }
 
+    /**
+     * Cheaper than calling {@code ctx.isRemoved()} from event loop
```

Comment:
We should tell what this method does first, from the standpoint of an API user:

```java
/**
 * Tells whether this handle has been removed from the {@link ChannelPipeline}.
 * This method is useful because 1) a user can check if this handler is removed even if
 * he or she does not have a reference to {@link ChannelHandlerContext} and 2) it is
 * cheaper than {@link ChannelHandlerContext#isRemoved()}.
 */
```